helix-core/src/main/java/org/apache/helix/controller/stages/TopStateHandoffReportStage.java

```diff
@@ -471,6 +474,7 @@ private void reportTopStateComesBack(ResourceControllerDataProvider cache, Map<S
       }
     }
 
+    // TODO: why are we checking if handoff happened within threshold here? it should be reported either way.
```

Comment:
If handoff happened within timeout, Helix treat it as regular handoff. 
If handoff happened longer than timeout, Helix treat it as handoff failure. I think there may be different metrics counting this. Please read it though and see how the new metrics fit into picture. 
I would prefer not leave question as comments in code.

Reply:
Sure let's discuss this over above filed issue. I removed this TODO because this is different thread of discussion and I don't want to hold this PR until we conclude on above topic. Thanks for the suggestion!